app/code/Magento/MediaGalleryUiApi/Api/GetInsertImageDataInterface.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryUiApi\Api;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * Class responsible to provide insert image details
+ */
+interface GetInsertImageDataInterface extends ExtensibleDataInterface
+{
+    /**
+     * Retrieves a content (just a link or an html block) for inserting image to the content
+     *
+     * @param string $encodedFilename
+     * @param bool $forceStaticPath
+     * @param bool $renderAsTag
+     * @param int|null $storeId
+     * @return null|string
+     */
+    public function getImageContent(
```

Comment:
This method should be part of the service, it shouldn't be present in data transfer object https://en.wikipedia.org/wiki/Data_transfer_object